tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -1177,6 +1179,25 @@ def _rebuild_func(self, func):
         func.structured_input_signature)
     return rebuilt_func
 
+  def _execute_calibration(self, calibration_input_fn):
+    for inp in calibration_input_fn():
+      if isinstance(inp, dict):
+        self._converted_func(
+            **{k: ops.convert_to_tensor(v) for k, v in inp.items()})
+      else:
+        self._converted_func(*map(ops.convert_to_tensor, inp))
+
+    def _save_calibration_table(node):
+      calibration_table = gen_trt_ops.get_calibration_data_op(
+          _get_canonical_engine_name(node.name))
+      node.attr["calibration_data"].s = calibration_table.numpy()
+
+    self._for_each_trt_node(self._converted_graph_def,
+                            _save_calibration_table)
+
+    # Rebuild the function since calibration has changed the graph.
+    self._converted_func = self._rebuild_func(self._converted_func)
+
   # TODO(laigd): provide a utility function to optimize a ConcreteFunction and
   # use it here (b/124792963).
   def convert(self, calibration_input_fn=None):
```

Comment:
Can we add description to calibration_input_fn to document the behavior, such as we run calibration in this step only for non-dynamic shape mode. Otherwise, we record the input_fn and delay the calibration until build time?